src/overlays/actors/ovl_En_Vali/z_en_vali.c

```diff
@@ -76,71 +82,732 @@ static ColliderCylinderInit D_80B288F0 = {
+            EnVali_SetupMoveArmsDown(this);
+        } else {
+            EnVali_SetupDivideAndDie(this, globalCtx);
+        }
+    }
+}
+
+void EnVali_MoveArmsDown(EnVali* this, GlobalContext* globalCtx) {
+    if (SkelAnime_Update(&this->skelAnime)) {
+        EnVali_SetupFloatIdle(this);
+    }
+}
+
+void EnVali_Burnt(EnVali* this, GlobalContext* globalCtx) {
+    if (this->timer != 0) {
+        this->timer--;
+    }
+
+    if (this->timer == 0) {
+        EnVali_SetupDivideAndDie(this, globalCtx);
+    }
+}
+
+void EnVali_DivideAndDie(EnVali* this, GlobalContext* globalCtx) {
+    static Vec3f effectVelocity = { 0.0f, 0.0f, 0.0f };
+    static Vec3f effectAccel = { 0.0f, 0.0f, 0.0f };
+    s16 effectScale;
+    Vec3f effectPos;
+    s32 i;
+
+    if (this->timer != 0) {
+        this->timer--;
+    }
+
+    for (i = 0; i < 2; i++) {
+        effectPos.x = this->actor.world.pos.x + Rand_CenteredFloat(20.0f);
+        effectPos.y = this->actor.world.pos.y + Rand_CenteredFloat(8.0f);
+        effectPos.z = this->actor.world.pos.z + Rand_CenteredFloat(20.0f);
+        effectVelocity.y = (Rand_ZeroOne() + 1.0f);
+        effectScale = Rand_S16Offset(40, 40);
+
+        if (Rand_ZeroOne() < 0.7f) {
+            EffectSsDtBubble_SpawnColorProfile(globalCtx, &effectPos, &effectVelocity, &effectAccel, effectScale, 25, 2,
+                                               1);
+        } else {
+            EffectSsDtBubble_SpawnColorProfile(globalCtx, &effectPos, &effectVelocity, &effectAccel, effectScale, 25, 0,
+                                               1);
+        }
+    }
+
+    if (this->timer == 0) {
+        Actor_Kill(&this->actor);
+    }
+}
+
+void EnVali_Stunned(EnVali* this, GlobalContext* globalCtx) {
+    SkelAnime_Update(&this->skelAnime);
+
+    if (this->timer != 0) {
+        this->timer--;
+    }
+
+    if (this->actor.velocity.y != 0.0f) {
+        if (Math_StepToF(&this->actor.world.pos.y, this->actor.floorHeight, this->actor.velocity.y)) {
+            this->actor.velocity.y = 0.0f;
+            Audio_PlayActorSound2(&this->actor, NA_SE_EN_DODO_M_GND);
+        } else {
+            this->actor.velocity.y += 1.0f;
+        }
+    }
+
+    if (this->timer == 0) {
+        this->bodyCollider.info.bumper.effect = 1; // Shock?
+        EnVali_SetupFloatIdle(this);
+    }
+}
+
+void EnVali_Frozen(EnVali* this, GlobalContext* globalCtx) {
+    Vec3f effectPos;
+    s32 temp_v0;
+    s32 temp_v1;
+
+    if (this->timer != 0) {
+        this->timer--;
+    }
+
+    temp_v1 = this->timer - 20;
+    this->actor.colorFilterTimer = 36;
+
+    if (temp_v1 > 0) {
+        temp_v0 = temp_v1 >> 1;
+
+        if ((this->timer & 1) != 0) {
+            effectPos.y = this->actor.world.pos.y - 20.0f + (-temp_v0 * 5 + 40);
+            // Cannot be %
+            effectPos.x = this->actor.world.pos.x + (((temp_v0 & 2) != 0) ? 12.0f : -12.0f);
+            effectPos.z = this->actor.world.pos.z + (((temp_v0 & 1) != 0) ? 12.0f : -12.0f);
```

Comment:
does this not work?
```suggestion
            effectPos.x = this->actor.world.pos.x + ((temp_v0 & 2) ? 12.0f : -12.0f);
            effectPos.z = this->actor.world.pos.z + ((temp_v0 & 1) ? 12.0f : -12.0f);
```

Reply:
It does, but the former can be 2, so I didn't think it was appropriate to suppress the `!= 0`.